### PR TITLE
fix tests with invalid vfs.s3.no_sign_request setting

### PR DESCRIPTION
**Pull request by atolopko-czi**
- reinstate `vfs.s3.no_sign_request=True` setting to allow these tests to pass on EC2 instances that have an IAM role and that does not grant explicit access to the bucket
- remove one test, `test_open_soma_with_plain_soma_context()` that can never be successfully run under these circumstances; there is still sufficient coverage from other tests
- tweak test assertions

Tests passing on GHA standard runner and on EC2 instance with IAM role, where tests previously failed.  Relevant tests pass on GHA custom runner: test run [here](https://github.com/chanzuckerberg/cellxgene-census/actions/runs/7266987244)

### CODE REVIEW

File: api/python/cellxgene_census/tests/test_open.py [hunk 1]
```diff
@@ -63,34 +61,36 @@ def test_open_soma_with_customized_plain_soma_context(latest_locator: CensusLoca
...
     with cellxgene_census.open_soma(census_version="latest", context=custom_context) as census:
-        # Verify the soma context defaults are set correctly in the TileDB context object.
-        assert census.context.tiledb_ctx.config()["vfs.s3.no_sign_request"] == "false"
+        # Verify the non-overriden soma context defaults are set correctly in the TileDB context object.
+        assert census.context.tiledb_ctx.config()["vfs.s3.no_sign_request"] == "true"
         assert census.context.tiledb_ctx.config()["vfs.s3.region"] == latest_locator.get("s3_region")
+        assert census.context.tiledb_ctx.config()["py.init_buffer_bytes"] == f"{1 * 1024 ** 3}"
 
-        # Verify that the user-provided custom config is set correctly in the TileDB context object.
+        # Verify that the user-overriden config settings are set correctly in the TileDB context object.
```
Comment (bkmartinjr) [inline api/python/cellxgene_census/tests/test_open.py:93]: typo in overriden (missing 'd')

**Review by bkmartinjr — APPROVED** [review]

File: api/python/cellxgene_census/tests/test_open.py (hunk 1 above)
Comment (atolopko-czi) [reply api/python/cellxgene_census/tests/test_open.py]: ty fixed

**Review by ebezzi — APPROVED** [review]
LGTM. @atolopko-czi, with this new paradigm, I would recommend to change `test_opening_census_without_anon_access_fails_with_bogus_creds` to use a context where the unsigned request is disabled, instead of using `context=soma.SOMATileDBContext()`.

**Comment by atolopko-czi** [comment]
> LGTM. @atolopko-czi, with this new paradigm, I would recommend to change `test_opening_census_without_anon_access_fails_with_bogus_creds` to use a context where the unsigned request is disabled, instead of using `context=soma.SOMATileDBContext()`.

It's a good idea, but it's hard to make that a reliable test, since it depends upon the test environment and whether there is an IAM role configured. See discussion [here](https://github.com/chanzuckerberg/cellxgene-census/issues/899#issuecomment-1863163031), as it would take implementing one of Bruce's suggestions there. It could be worth a separate issue, I suppose. But I think it's more than we should implement on this particular PR.

**Comment by bkmartinjr** [comment]
>  a separate issue

definitely agree that we should land this PR and unblock the acceptance tests, then fine tune

**Pull request merged**